### PR TITLE
948: refactor sub-milestones to use fullname field

### PR DESCRIPTION
**Pull request by trbmcginnis**
- Refactor sub-milestones to use `fullname` field (which is originally sourced from a left join with the contact table in the API)
- `fullname` takes the place of `dcpRecommendationsubmittedbyname`, which is oftentimes NULL in the database
- `fullname` is used to associate a specific disposition with a milestone AND to display which user submitted a hearing/vote
- there are already tests that make sure that nothing breaks when `fullname` is NULL in the database

Addresses #948

### CODE REVIEW

File: app/components/hearings-list-for-milestones-list.js [hunk 1]
```diff
@@ -68,23 +68,18 @@ export default class HearingsListForMilestonesListComponent extends Component {
...
-          filteredDispositionsArray.push(disposition);
-        }
-      }
+    const currentMilestoneDispositions = dispositions.filter(function(disposition) {
+      // make sure we aren't grabbing any null values; null values break substring
+      const participantType = disposition.fullname != null ? disposition.fullname.substring(3, 5) : '';
+      return milestoneParticipantReviewLookup[milestone.displayName] === participantType;
     });
-    return filteredDispositionsArray;
+    return currentMilestoneDispositions;
```
Comment (allthesignals) [inline app/components/hearings-list-for-milestones-list.js]: _great_ — you can even go a step further and just return the `dispositions.filter`.

File: app/components/hearings-list-for-milestones-list.js [hunk 2]
```diff
@@ -111,16 +106,15 @@ export default class HearingsListForMilestonesListComponent extends Component {
     // This participantRecommendationType is passed into `deduped-votes-list` to assure that we are deduplicating
     // by the correct recommenation field.
     const milestoneParticipants = currentMilestoneDispositions.map(disp => ({
-      landUseParticipantFullName: disp.recommendationSubmittedByFullName,
+      landUseParticipantFullName: disp.fullNameLongFormat,
       participantRecommendationType: partRecType, // e.g. 'dcpCommunityboardrecommendation'
-      landUseParticipantType: disp.dcpRecommendationsubmittedbyname.substring(2, 4), // e.g. 'CB'
+      landUseParticipantType: disp.fullname.substring(3, 5), // e.g. 'CB'
```
Comment (allthesignals) [inline app/components/hearings-list-for-milestones-list.js]: might wanna comment this logic more

Comment (trbmcginnis) [reply app/components/hearings-list-for-milestones-list.js]: will do!

File: app/components/hearings-list-for-milestones-list.js (hunk 1 above)
Comment (trbmcginnis) [reply app/components/hearings-list-for-milestones-list.js]: will do :)

**Pull request merged**